core/src/main/java/com/linecorp/armeria/common/HttpHeaders.java

```diff
@@ -13,227 +13,157 @@
-     */
-    @Nullable
-    String authority();
-
-    /**
-     * Sets the {@link HttpHeaderNames#AUTHORITY} header.
-     */
-    HttpHeaders authority(String authority);
-
-    /**
-     * Gets the {@link HttpHeaderNames#PATH} header or {@code null} if there is no such header.
-     */
-    @Nullable
-    String path();
-
-    /**
-     * Sets the {@link HttpHeaderNames#PATH} header.
-     */
-    HttpHeaders path(String path);
-
-    /**
-     * Gets the {@link HttpHeaderNames#STATUS} header or {@code null} if there is no such header.
-     */
-    @Nullable
-    HttpStatus status();
-
-    /**
-     * Sets the {@link HttpHeaderNames#STATUS} header.
-     */
-    HttpHeaders status(int statusCode);
-
-    /**
-     * Sets the {@link HttpHeaderNames#STATUS} header.
-     */
-    HttpHeaders status(HttpStatus status);
-
-    /**
-     * Returns the value of the {@code 'content-type'} header.
-     * @return the valid header value if present. {@code null} otherwise.
+     * Returns a new {@link HttpHeaders} with the specified headers.
      */
-    @Nullable
-    MediaType contentType();
-
-    /**
-     * Sets the {@link HttpHeaderNames#CONTENT_TYPE} header.
-     */
-    HttpHeaders contentType(MediaType mediaType);
-
-    /**
-     * Copies the entries missing in this headers from the specified {@link Headers}.
-     * This method is a shortcut of the following code:
-     * <pre>{@code
-     * headers.names().forEach(name -> {
-     *      if (!contains(name)) {
-     *          set(name, headers.getAll(name));
-     *      }
-     * });
-     * }</pre>
-     */
-    default HttpHeaders setAllIfAbsent(Headers<AsciiString, String, ?> headers) {
-        requireNonNull(headers, "headers");
-        if (!headers.isEmpty()) {
-            headers.names().forEach(name -> {
-                if (!contains(name)) {
-                    set(name, headers.getAll(name));
-                }
-            });
-        }
-        return this;
+    static HttpHeaders of(CharSequence name1, CharSequence value1,
+                          CharSequence name2, CharSequence value2,
+                          CharSequence name3, CharSequence value3,
+                          CharSequence name4, CharSequence value4) {
+        return new HttpHeadersBuilder().add(name1, value1)
+                                       .add(name2, value2)
+                                       .add(name3, value3)
+                                       .add(name4, value4)
+                                       .build();
     }
 
     /**
-     * Returns the immutable view of this headers.
+     * Returns a new {@link HttpHeaders} with the specified headers. The values are converted into
+     * {@link String}s via {@link HttpHeadersBase#addObject(CharSequence, Object)}.
      */
-    default HttpHeaders asImmutable() {
-        return new ImmutableHttpHeaders(this);
+    static HttpHeaders of(CharSequence name1, Object value1,
```

Comment:
How about also `CharSequence... nameValues` and `checkArgument(nameValues.length % 2 == 0)`?

Reply:
If we change the value type to `String`, do we perform an automatic conversion to `String` for the values? (which is fine by me.)

Reply:
Ah sorry I might have looked at an old version or just got confused, I thought the overrides were using `CharSequence` for the `value`. If not, then this is already best.